File: CODIGOS6/104.swift
import Foundation

func main() {
  let n = Int(readLine()!)!
  var eggs = [Int](repeating: 0, count: 20)
  var touching = [[Int]](repeating: [Int](repeating: 0, count: 20), count: 20)
  var ans = [Int]()
  for i in 0..<n {
    cin >> eggs[i]
  }
  for i in 0..<n {
    for j in 0..<n {
      touching[i][j] = 0
    }
  }
  for i in 0..<n {
    var k = Int(readLine()!)!
    for j in 0..<k {
      let t = Int(readLine()!)!
      touching[i][t] = 1
    }
  }
  var cnt = 0
  for i in 0..<n {
    var flag = true
    for j in 0..<n {
      if eggs[j] == 1 && touching[i][j] == 1 {
        flag = false
        break
      }
    }
    if flag {
      ans.append(i)
    }
  }
  for i in 0..<ans.count {
    print(ans[i])
  }
}

File: CODIGOS6/104.go
package main

import (
  "bufio"
  "fmt"
  "os"
)

func main() {
  scanner := bufio.NewScanner(os.Stdin)
  n := scanner.Scan()
  if !n {
    return
  }
  eggs := make([]int, 20)
  for i := 0; i < 20; i++ {
    eggs[i] = scanner.Int()
  }
  touching := make([][]int, 20)
  for i := 0; i < 20; i++ {
    touching[i] = make([]int, 20)
  }
  for i := 0; i < 20; i++ {
    n := scanner.Scan()
    if !n {
      return
    }
    for j := 0; j < n; j++ {
      touching[i][scanner.Int()] = 1
    }
  }
  ans := []int{}
  for i := 0; i < 20; i++ {
    flag := true
    for j := 0; j < 20; j++ {
      if eggs[j] == 1 && touching[i][j] == 1 {
        flag = false
        break
      }
    }
    if flag {
      ans = append(ans, i)
    }
  }
  for i := 0; i < len(ans); i++ {
    fmt.Println(ans[i])
  }
}
